src/khepri.erl

```diff
@@ -1278,7 +1317,448 @@ count(PathPattern, Options) when is_map(Options) ->
+      StoreId :: khepri:store_id(),
+      PathPattern :: khepri_path:pattern(),
+      Fun :: khepri:fold_fun(),
+      Acc :: khepri:fold_acc(),
+      Ret :: khepri:ok(NewAcc) | khepri:error(),
+      NewAcc :: Acc;
+(PathPattern, Fun, Acc, Options) -> Ret when
+      PathPattern :: khepri_path:pattern(),
+      Fun :: khepri:fold_fun(),
+      Acc :: khepri:fold_acc(),
+      Options :: khepri:query_options() | khepri:tree_options(),
+      Ret :: khepri:ok(NewAcc) | khepri:error(),
+      NewAcc :: Acc.
+%% @doc Calls `Fun' on successive tree nodes matching the given path pattern,
+%% starting with `Acc'.
+%%
+%% This function accepts the following two forms:
+%% <ul>
+%% <li>`fold(StoreId, PathPattern, Fun, Acc)'. Calling it is the same as
+%% calling `fold(StoreId, PathPattern, Fun, Acc, #{})'.</li>
+%% <li>`fold(PathPattern, Fun, Acc, Options)'. Calling it is the same as
+%% calling `fold(StoreId, PathPattern, Fun, Acc, Options)' with the default
+%% store ID (see {@link khepri_cluster:get_default_store_id/0}).</li>
+%% </ul>
+%%
+%% @see fold/5.
+
+fold(StoreId, PathPattern, Fun, Acc) when ?IS_STORE_ID(StoreId) ->
+    fold(StoreId, PathPattern, Fun, Acc, #{});
+fold(PathPattern, Fun, Acc, Options) when is_map(Options) ->
+    StoreId = khepri_cluster:get_default_store_id(),
+    fold(StoreId, PathPattern, Fun, Acc, Options).
+
+-spec fold(StoreId, PathPattern, Fun, Acc, Options) -> Ret when
+      StoreId :: khepri:store_id(),
+      PathPattern :: khepri_path:pattern(),
+      Fun :: khepri:fold_fun(),
+      Acc :: khepri:fold_acc(),
+      Options :: khepri:query_options() | khepri:tree_options(),
+      Ret :: khepri:ok(NewAcc) | khepri:error(),
+      NewAcc :: Acc.
+%% @doc Calls `Fun' on successive tree nodes matching the given path pattern,
+%% starting with `Acc'.
+%%
+%% The `PathPattern' can be provided as a native path pattern (a list of tree
+%% node names and conditions) or as a string. See {@link
+%% khepri_path:from_string/1}.
+%%
+%% Like the function passed to {@link maps:fold/3}, `Fun' must accept the
+%% following arguments:
+%% <ol>
+%% <li>the native path of the tree node being handled</li>
+%% <li>the tree node properties map</li>
+%% <li>an Erlang term which is either `Acc' for the first matched tree node, or
+%% the return value of the previous call to `Fun'</li>
+%% </ol>
+%%
+%% The returned `{ok, NewAcc}' tuple contains the return value of the last call
+%% to `Fun', or `Acc' if no tree nodes matched the given path pattern.
+%%
+%% Example: list all nodes in the tree
+%% ```
+%% %% List all tree node paths in the tree. The tree is:
+%% %% <root>
+%% %% `-- foo
+%% %%     `-- bar = value
+%% {ok, [[foo], [foo, bar]]}} = khepri:fold(
+%%                                  StoreId,
+%%                                  fun(Path, _NodeProps, Acc) ->
+%%                                      [Path | Acc]
+%%                                  end, [],
+%%                                  [?KHEPRI_WILDCARD_STAR_STAR]).
```

Comment:
I think the PathPattern and Fun arguments should be swapped here, right?

Reply:
It looks like the same applies to the code examples for the other new functions too

Reply:
You're right, thanks!

I miss the testability of code snippets in the documentation, like with Rust.